### DISCUSSION TITLE
How to send a HashMap from Rust side to Javascript as Map instead of Object?

### DISCUSSION THREAD

**Discussion by CrendKing** [tauri-apps/tauri#9153] [Q&A] [answered]
I noticed if a Rust command returns a HashMap<K, V>, it is received in Javascript as Object. Is there a way to change it to Javascript Map<K, V> instead? What code controls the transformation? Thank you!

**Comment by CrendKing** [comment]
Did some search and my findings are:

1) The returned data from Rust command is serialized to JSON by serde_json.
2) The data is sent to Javascript side via HTTP request.
3) Upon receiving the response, the IPC code calls [`response.json()`](https://github.com/tauri-apps/tauri/blob/d349558abbefc78baf98f7837e5237f370758931/core/tauri/scripts/ipc-protocol.js#L40), which automatically deserializes the JSON object into Javascript object.

It seems we have to manually create a new Map from the object in the `invoke` result (e.g. `new Map(Object.entries(value))`)?

**Reply by FabianLars** [reply, marked as answer]
> It seems we have to manually create a new Map from the object in the invoke result (e.g. new Map(Object.entries(value)))?

yep
